include/cuco/utility/key_generator.cuh

```diff
@@ -414,4 +420,90 @@ class key_generator {
   RNG rng_;  ///< Random number generator
 };
 
+/**
+ * @brief Generates sequences of random byte strings with random lengths
+ *
+ * @tparam RNG Pseudo-random number generator
+ *
+ * @throws If the minimum/maximum sequence lengths are implausible
+ *
+ * @param n_sequences Number of byte sequences to generate
+ * @param min_sequence_length Minimum sequence length
+ * @param max_sequence_length Maximum sequence length
+ * @param stream CUDA stream in which this operation is executed in
+ * @param seed Random seed
+ *
+ * @return A pair consisting of (1) a vector of spans pointing to each sequence and (2) a byte
+ * vector holding the actual data
+ */
+template <typename RNG = thrust::default_random_engine>
+std::pair<thrust::device_vector<cuda::std::span<std::byte>>, thrust::device_vector<std::byte>>
+generate_random_byte_sequences(std::size_t n_sequences,
+                               std::size_t min_sequence_length,
+                               std::size_t max_sequence_length,
+                               cudaStream_t stream = 0,
+                               std::size_t seed    = 0)
```

Comment:
```suggestion
                               std::size_t seed    = 0,
                               cudaStream_t stream = 0)
```
nit: Place stream the last argument following cudf's styling. question: Do we actually want to expose the stream parameter?

Reply:
It's not strictly neccessary since this function isn't async. However, I find it's best practice to always expose the stream parameter, in cases where the environment doesn't allow for using the default stream.